src/components/RangeSlider/RangeSlider.js

```diff
@@ -0,0 +1,246 @@
+   * and if in the upper half add 0 - 2.5%
+   * @function fixThumbRangeDeviation
+   * @param {number} hoverVal the almost accurate hover value
+   * @returns {number} returns the accurate value
+   */
+  const fixThumbRangeDeviation = hoverVal => {
+    const pos = getPositionInSlider(hoverVal)
+    let fixedValue = hoverVal
+    const fixRange = getFixRange()
+
+    const posFromCenter = (pos - 0.5) * 2
+    const adjustment = posFromCenter * fixRange
+    fixedValue += adjustment
+
+    if (step > 1) {
+      fixedValue = roundValueBasedOnStep(fixedValue)
+    }
+
+    if (fixedValue > max) fixedValue = max
+    if (fixedValue < min) fixedValue = min
+    return fixedValue
+  }
+
+  /**
+   * Calculates the middle of the range thumb. If relative position is less than
+   * half, it adds 0 - 2.5% to current pos, else - it subtracts 0 - 2.5%. With fixRangeRatio: 0 - 0.025%
+   * This function is about position of the thumb, the first function is about hover value.
+   * @function posTooltipMiddleThumb
+   * @param {number} val the value of the slider
+   * @returns {number} the middle position of the range thumb
+   */
+  const posTooltipMiddleThumb = val => {
+    const pos = getPositionInSlider(val)
+    let middleValue = Number(val)
+    const fixRange = getFixRange()
+
+    const posFromCenter = (pos - 0.5) / 0.5
+    const adjustment = posFromCenter * fixRange
+    middleValue -= adjustment
+
+    return middleValue
+  }
+
+  const calculatePercentage = e => {
+    const mouseX = Number(e.nativeEvent.offsetX)
+    const curHoverPos = (mouseX / e.target.clientWidth) * 100
+    const curHoverVal = getValueInSlider(curHoverPos)
+    if (curHoverVal > max || curHoverVal < min) return
+    const stepDecimalCount = countDecimals()
+
+    const updatedVal = fixThumbRangeDeviation(curHoverVal)
+    setHoverPos(curHoverPos)
+    setHoverValue(updatedVal.toFixed(stepDecimalCount))
+  }
+
+  const posTooltipToHover = e => {
+    if (disabled) return
+    calculatePercentage(e)
+    isToggleTooltip && setShowTooltip(true)
+  }
+
+  const hideTooltip = () => {
+    setShowTooltip(false)
+  }
+
+  const posTooltipToValue = () => {
+    if (disabled) return
+    if (isToggleTooltip) return hideTooltip()
+    const middleValue = posTooltipMiddleThumb(value)
+    setHoverPos(100 * getPositionInSlider(middleValue))
+    setHoverValue(value)
+  }
+
+  const renderLabels = () => {
+    const minLabel = classNames(styles.label, styles.minLabel)
+    const maxLabel = classNames(styles.label, styles.maxLabel)
+
+    return (
+      <>
+        <span className={minLabel}>{min}</span>
+        <span className={maxLabel}>{max}</span>
+      </>
+    )
+  }
+
+  const renderTooltip = () => {
+    const tooltipEl = (
+      <div ref={tooltipRef} className={styles.tooltip}>
+        {hoverValue ?? value}
+      </div>
+    )
+
+    if (isToggleTooltip) return showTooltip && tooltipEl
+    return tooltipEl
+  }
```

Comment:
```suggestion
  const renderTooltip = () => (!isToggleTooltip || showTooltip) ? (
    <div ref={tooltipRef} className={styles.tooltip}>
      {hoverValue ?? value}
    </div>
  ) : null
  }
```
I think it has the same result - check it and choose what you prefer